crates/red_knot_python_semantic/src/types.rs

```diff
@@ -909,6 +931,10 @@ impl<'db> Type<'db> {
                 Type::Callable(CallableType::General(target_callable)),
             ) => self_callable.is_assignable_to(db, target_callable),
 
+            (Type::Instance(InstanceType { class: self_class }), Type::Callable(_)) => {
+                Type::ClassLiteral(ClassLiteralType { class: self_class }).is_subtype_of(db, target)
```

Comment:
As discussed above, this delegation isn't correct. A class object does not have the same call signature as an instance of that class. This branch should handle the checking of `__call__` method.